### PR TITLE
Added MetaPlAn subworkflow

### PR DESCRIPTION
**Pull request by LaurenGonsalves**
## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/main.nf [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    ch_versions = Channel.empty()
+
+    METAPHLAN_MAKEDB ( )
+    ch_versions = ch_versions.mix(METAPHLAN_MAKEDB.out.versions.first())
+
+    METAPHLAN_METAPHLAN ( ch_fastq, METAPHLAN_MAKEDB.out.db )
+    ch_versions = ch_versions.mix(METAPHLAN_METAPHLAN.out.versions.first())
+
+    METAPHLAN3_MERGEMETAPHLANTABLES ( METAPHLAN_METAPHLAN.out.profile.map{ [ [id:'all_samples'], it[1] ] }.groupTuple( sort: 'deep') )
```
Comment (CarsonJM) [inline subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/main.nf:21]:
```suggestion
    METAPHLAN_MERGEMETAPHLANTABLES ( METAPHLAN_METAPHLAN.out.profile.map{ [ [id:'all_samples'], it[1] ] }.groupTuple( sort: 'deep') )
```

File: subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/main.nf [hunk 2]
```diff
@@ -0,0 +1,30 @@
+include { METAPHLAN_MAKEDB      } from '../../../modules/nf-core/metaphlan/makedb/main'
+include { METAPHLAN_METAPHLAN     } from '../../../modules/nf-core/metaphlan/metaphlan/main'
+include { METAPHLAN3_MERGEMETAPHLANTABLES     } from '../../../modules/nf-core/metaphlan3/mergemetaphlantables/main'
```
Comment (CarsonJM) [inline subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/main.nf:3]:
```suggestion
include { METAPHLAN_MERGEMETAPHLANTABLES     } from '../../../modules/nf-core/metaphlan/mergemetaphlantables/main'
```

File: subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/main.nf [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    ch_versions = ch_versions.mix(METAPHLAN_MAKEDB.out.versions.first())
+
+    METAPHLAN_METAPHLAN ( ch_fastq, METAPHLAN_MAKEDB.out.db )
+    ch_versions = ch_versions.mix(METAPHLAN_METAPHLAN.out.versions.first())
+
+    METAPHLAN3_MERGEMETAPHLANTABLES ( METAPHLAN_METAPHLAN.out.profile.map{ [ [id:'all_samples'], it[1] ] }.groupTuple( sort: 'deep') )
+    ch_versions = ch_versions.mix(METAPHLAN3_MERGEMETAPHLANTABLES.out.versions.first())
+
+    emit:
+    merged_taxa      = METAPHLAN3_MERGEMETAPHLANTABLES.out.txt
```
Comment (CarsonJM) [inline subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/main.nf:25]:
```suggestion
    merged_taxa      = METAPHLAN_MERGEMETAPHLANTABLES.out.txt
```

File: subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/main.nf [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    ch_versions = Channel.empty()
+
+    METAPHLAN_MAKEDB ( )
+    ch_versions = ch_versions.mix(METAPHLAN_MAKEDB.out.versions.first())
+
+    METAPHLAN_METAPHLAN ( ch_fastq, METAPHLAN_MAKEDB.out.db )
+    ch_versions = ch_versions.mix(METAPHLAN_METAPHLAN.out.versions.first())
+
+    METAPHLAN3_MERGEMETAPHLANTABLES ( METAPHLAN_METAPHLAN.out.profile.map{ [ [id:'all_samples'], it[1] ] }.groupTuple( sort: 'deep') )
+    ch_versions = ch_versions.mix(METAPHLAN3_MERGEMETAPHLANTABLES.out.versions.first())
```
Comment (CarsonJM) [inline subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/main.nf:22]:
```suggestion
    ch_versions = ch_versions.mix(METAPHLAN_MERGEMETAPHLANTABLES.out.versions.first())
```

File: subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/meta.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  - metagenomics
+  - database
+  - classification
+  - merge
+  - profiles
+components:
+  - metaphlan/makedb
+  - metaphlan/metaphlan
+  - metaphlan/mergemetaphlantables
+input:
```
Comment (CarsonJM) [inline subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/meta.yml:14]:
```suggestion

input:
```
Personal preference, I'll leave it to you

File: subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/tests/main.nf.test [hunk 6]
```diff
@@ -0,0 +1,53 @@
+// nf-core subworkflows test fastq_taxonomic_profile_metaphlan
```
Comment (CarsonJM) [inline subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/tests/main.nf.test:1]:
```suggestion
```

File: subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/tests/main.nf.test [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    tag "metaphlan/metaphlan"
+    tag "metaphlan3/mergemetaphlantables"
+
+
+    test("sarscov2 - fastq - paired_end") {
+
+        when {
+            workflow {
+                """
+                // TODO nf-core: define inputs of the workflow here.
```
Comment (CarsonJM) [inline subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/tests/main.nf.test:22]:
```suggestion
```

File: subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/tests/nextflow.config [hunk 8]
```diff
@@ -0,0 +1,9 @@
+process {
+
+    publishDir = { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" }
+
```
Comment (CarsonJM) [inline subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/tests/nextflow.config:4]:
```suggestion
```

File: subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/tests/nextflow.config [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+    withName: METAPHLAN_MAKEDB {
+        ext.args = '--index mpa_vJan21_TOY_CHOCOPhlAnSGB_202103'
+    }
+
```
Comment (CarsonJM) [inline subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/tests/nextflow.config:8]:
```suggestion
```

**Review by CarsonJM — CHANGES_REQUESTED** [review]
Nice! Great work! Just a few changes

File: subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/tests/main.nf.test [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+                )
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert workflow.success},
+                { assert snapshot(workflow.out).match()}
+                //TODO nf-core: Add all required assertions to verify the test output.
```
Comment (maxulysse) [inline subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/tests/main.nf.test:45]:
```suggestion
```

File: subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/tests/nextflow.config [hunk 11]
```diff
@@ -0,0 +1,6 @@
+process {
+
```
Comment (maxulysse) [inline subworkflows/nf-core/fastq_taxonomic_profile_metaphlan/tests/nextflow.config:2]:
```suggestion
```

**Review by maxulysse — APPROVED** [review]
Minor suggestions

**Review by maxulysse — APPROVED** [review]

**Comment by maxulysse** [comment]
I can see that `all_samples.txt` doesn't have a common md5 between conda and docker, can you check the content?
I'm assuming that some specific lines would be specific to conda, so either a path to the lib, in which case, I'd suggest snapshotting some of this file's lines, or maybe a different version of a dependency, in which case, I'd add this as a dependency in the environment.yml, similarly to what we've done already with samtools

**Review by CarsonJM — APPROVED** [review]

**Pull request merged**